src/api/ledger/ledger.js

```diff
@@ -0,0 +1,26 @@
+/* @flow */
+'use strict';
+const utils = require('./utils');
+const validate = utils.common.validate;
+const composeAsync = utils.common.composeAsync;
+const parseLedger = require('./parse/ledger');
+
+function getLedgerAsync(options, callback) {
+  validate.getLedgerOptions(options);
+
+  const request = {
+    ledger: options.ledgerVersion || 'validated',
+    expand: options.includeAllData,
+    transactions: options.includeTransactions,
+    accounts: options.includeAccounts
+  };
+
+  this.remote.requestLedger(request,
+    composeAsync(response => parseLedger(response.ledger), callback));
+}
+
+function getLedger(options={}) {
+  return utils.promisify(getLedgerAsync.bind(this))(options);
```

Comment:
Not sure if I agree it's more readable, I actually kind of like having the context binding separated from the function arguments. Regarding optimization, I tend to agree with the quote: "We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil”. http://www.catb.org/esr/writings/taoup/html/ch01s06.html#ftn.id2878872

In this case, I don't think it's a problem, so I'll change it since you both prefer it.